packages/EasyCore/tests/Bridge/Symfony/Valiator/Constraints/AbnValidatorTest.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Tests\Bridge\Symfony\Valiator\Constraints;
+
+use EoneoPay\Utils\Exceptions\UnexpectedTypeException;
+use EonX\EasyCore\Bridge\Symfony\Validator\Constraints\Abn;
+use EonX\EasyCore\Bridge\Symfony\Validator\Constraints\AbnValidator;
+use EonX\EasyCore\Tests\Bridge\Symfony\AbstractSymfonyTestCase;
+use Mockery\MockInterface;
+use stdClass;
+use Symfony\Component\Validator\Constraints\NotNull;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+/**
+ * @covers \EonX\EasyCore\Bridge\Symfony\Validator\Constraints\AbnValidator
+ */
+final class AbnValidatorTest extends AbstractSymfonyTestCase
+{
+    /**
+     * @return mixed[]
+     */
+    public function provideAbnWithInvalidCharacters(): array
+    {
+        return [
+            'Starts with zero' => ['00043145470'],
+            'Contains letter' => ['1234567890a'],
+            'Contains special symbols' => ['1234(678900'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideUnexpectedValues(): array
+    {
+        return [
+            'class without __toString' => [new stdClass(), 'Expected argument of type "string", "stdClass" given'],
+            'integer' => [53004085616, 'Expected argument of type "string", "integer" given'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideValidABN(): array
```

Comment:
please, use `Abn` instead of `ABN`. Here and below.

Reply:
Done.